js/src/structured-data-blocks/how-to/components/HowTo.js

```diff
@@ -56,11 +56,8 @@ export default class HowTo extends Component {
 		this.toggleListType  = this.toggleListType.bind( this );
 		this.setDurationText = this.setDurationText.bind( this );
 
-		if ( ! this.props.attributes.durationText ) {
-			const defaultDurationText = this.getDefaultDurationText();
-
-			this.setDurationText( defaultDurationText );
-		}
+		const defaultDurationText = this.getDefaultDurationText();
+		this.setDefaultDurationText( defaultDurationText );
```

Comment:
unsure this should be done in the constructor, wouldn't be safer to do it in a lifecycle method e.g. componentDidMount?

Reply:
Yes, it is safe to do this in the constructor, because you do not depend on anything that is rendered. This also prevents a needless (virtual) rerender, because although teh component is output to the browser **after** `componentDidMount`, it will reconsolidate the virtual DOM.